github/resource_github_repository.go

```diff
@@ -815,45 +834,22 @@ func flattenSecurityAndAnalysis(securityAndAnalysis *github.SecurityAndAnalysis)
 		return []interface{}{}
 	}
 
-	advancedSecurityMap := make(map[string]interface{})
-	advancedSecurityMap["status"] = securityAndAnalysis.GetAdvancedSecurity().GetStatus()
-
-	secretScanningMap := make(map[string]interface{})
-	secretScanningMap["status"] = securityAndAnalysis.GetSecretScanning().GetStatus()
-
-	secretScanningPushProtectionMap := make(map[string]interface{})
-	secretScanningPushProtectionMap["status"] = securityAndAnalysis.GetSecretScanningPushProtection().GetStatus()
-
 	securityAndAnalysisMap := make(map[string]interface{})
-	securityAndAnalysisMap["advanced_security"] = []interface{}{advancedSecurityMap}
-	securityAndAnalysisMap["secret_scanning"] = []interface{}{secretScanningMap}
-	securityAndAnalysisMap["secret_scanning_push_protection"] = []interface{}{secretScanningPushProtectionMap}
-
-	return []interface{}{securityAndAnalysisMap}
-}
-
-func expandSecurityAndAnalysis(input []interface{}) *github.Repository {
-	if len(input) == 0 || input[0] == nil {
-		return nil
-	}
-
-	securityAndAnalysis := input[0].(map[string]interface{})
-	update := &github.SecurityAndAnalysis{}
 
-	advancedSecurity := securityAndAnalysis["advanced_security"].([]interface{})[0].(map[string]interface{})
-	update.AdvancedSecurity = &github.AdvancedSecurity{
-		Status: github.String(advancedSecurity["status"].(string)),
+	advancedSecurity := securityAndAnalysis.GetAdvancedSecurity()
+	if advancedSecurity != nil {
+		securityAndAnalysisMap["advanced_security"] = []interface{}{map[string]interface{}{
+			"status": advancedSecurity.GetStatus(),
+		}}
 	}
 
-	secretScanning := securityAndAnalysis["secret_scanning"].([]interface{})[0].(map[string]interface{})
-	update.SecretScanning = &github.SecretScanning{
-		Status: github.String(secretScanning["status"].(string)),
-	}
+	securityAndAnalysisMap["secret_scanning"] = []interface{}{map[string]interface{}{
+		"status": securityAndAnalysis.GetSecretScanning().GetStatus(),
+	}}
 
-	secretScanningPushProtection := securityAndAnalysis["secret_scanning_push_protection"].([]interface{})[0].(map[string]interface{})
-	update.SecretScanningPushProtection = &github.SecretScanningPushProtection{
-		Status: github.String(secretScanningPushProtection["status"].(string)),
-	}
+	securityAndAnalysisMap["secret_scanning_push_protection"] = []interface{}{map[string]interface{}{
+		"status": securityAndAnalysis.GetSecretScanningPushProtection().GetStatus(),
+	}}
 
-	return &github.Repository{SecurityAndAnalysis: update}
+	return []interface{}{securityAndAnalysisMap}
```

Comment:
A concrete type could be introduced so that there is one, but the `flattenSecurityAndAnalysis` function return type [previously was `[]interface{}`](https://github.com/integrations/terraform-provider-github/blob/f81d0215f860d49e47bd945ddc788c3856c58bc6/github/resource_github_repository.go#L813-L833), so I kept it the same when I was refactoring it / making the properties optional.

I also removed the [`expandSecurityAndAnalysis` function](https://github.com/integrations/terraform-provider-github/blob/f81d0215f860d49e47bd945ddc788c3856c58bc6/github/resource_github_repository.go#L835-L859). The diff is lining the changes with `flattenSecurityAndAnalysis` up with the `expandSecurityAndAnalysis` removal.